nipype/pipeline/engine/utils.py

```diff
@@ -455,7 +466,7 @@ def format_node(node, format='python', include_config=False):
     return lines
 
 
-def modify_paths(object, relative=True, basedir=None):
+def modify_paths(object, relative=True, basedir=None, type_hints=None):
```

Comment:
@effigies - you may want to take a peek at the recursive inputs hash function in core specs. since the hints tell whether to hash a path (nested or not) or not, i think that function may be relevant in this scenario.